src/System.Text.RegularExpressions/src/System/Text/ValueStringBuilder.Reverse.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text
+{
+    internal ref partial struct ValueStringBuilder
+    {
+        public void AppendReversed(ReadOnlySpan<char> value)
+        {
+            Span<char> span = AppendSpan(value.Length);
+            for (int i = 0; i < value.Length; i++)
```

Comment:
If you use `span.Length` as the condition, the JIT is more likely to eliminate the bounds check

Reply:
Isn't the bounds check happening for both span (writing) and value (reading)? I expected that when using value.Length the value bounds check will eliminated and when using span.Length the span bounds check is eliminated.

Reply:
Makes sense. Thanks Jan